src/dynamo0p3.py

```diff
@@ -2849,9 +3193,11 @@ def scalar(self, scalar_arg):
     def fs_compulsory(self, function_space):
         '''add compulsory arguments common to operators and
         fields on a space.'''
-        # There is currently one compulsory argument: "ndf".
-        ndf_name = get_fs_ndf_name(function_space)
-        self._arglist.append(ndf_name)
+        if self._kern.cma_operation not in ["matrix-matrix"]:
+            # There is currently one compulsory argument: "ndf" but only
```

Comment:
Good point :-) I've renamed it `fs_common` and done this for the base class too - 487524a.